arviz/plots/backends/matplotlib/pairplot.py

```diff
@@ -246,11 +245,19 @@ def plot_pair(
         point_estimate_marker_kwargs.setdefault("s", markersize + 50)
 
         if ax is None:
+            backend_kwargs.setdefault("sharex", "col")
```

Comment:
should we allow not sharing axes? or sharing the axis along a whole row with marginals?

Reply:
Both of these should be allowed by specifying `backend_kwargs`. In the first instance, it would be 
```python
backend_kwargs = {"sharex": False, "sharey": False}
```
and in the second instance
```python
backend_kwargs = {"sharey":"row"}
```

Reply:
yes, sorry I wasn't very clear. I am not sure making this possible benefits anyone, if a user sets share# to False, they will still not one set of ticklablels per plot yet it is supported and it makes ArviZ codebase bigger and harder to test and maintain (my main concern here). I think one could even pass `sharex="row"` if they wanted to see everything burn.

I'd personally hardcode sharing on a row/column basis to keep things simple as I think we all agree that should be the default, but I'm open to be convinced that maybe sharing set to True could also be useful and we might want to allow some extra freedom.

Side note, given how the code works and is tested we might want to keep this as is and defer to https://github.com/arviz-devs/GSoD/blob/main/2021/viz_evaluation.md on the question of "to what extent should bad practices be possible when using ArviZ?"

Reply:
OK, so just to be sure I understand, you're proposing two different options:
1. To hardcode axis sharing and ignore any sharing provided by the user (perhaps issuing a warning if sharex/sharey are provided in backend_kwargs to say that they will be ignored?)
2. To allow the user to set sharex=sharey=True and only that, so that all joint and marginal plots share the same axes? Perhaps through a new keyword? (I suggest a new keyword because sharey=True would break the marginal plots)
3. To close the PR and accept this bad practice? (I wasn't sure how to interpret "as is" in what you said)

I'd argue strongly against the third option, as at the moment it's impossible to produce plots that could be wrangled into publishable form with the code as is. If this is the desired option, I'd argue that the changes to L328 and L332 should be retained, because at least then the user can provide axes that are shared properly and not have the axis labels clobbered by the NullFormatter.

Reply:
Also side note, if we go down the opinionated route, I'd get rid of the y axis tick marks and labels on the marginal plots, but I didn't put that in here because it felt too opinionated to do without discussion.

Reply:
By "keeping this as is" I meant the PR, merging it without extra changes, definitely not keeping this bug. 

Regarding option 2, I would use `True` to not diverge too much from matplotlib. It is directly used when no marginals are present, and should be popped and handled correctly when marginals are present. As you say, sharey=True breaks the plot and should not be an option in my opinion.

Reply:
Do you have a preferred way forward here? Here are the changes I would make for each option:

1. If either sharex or sharey are provided, pop them and warn the user they'll be ignored
2. Allow sharex = True; if sharey = True is provided, allow it if marginal plots are off, otherwise share the y axis on all non-marginal plots
3. Make no changes to the PR

My reservations about the second option are that I'm not sure we should be going against default matplotlib behaviour; it could be counterintuitive if the user passes something expecting a certain output and arviz does something else, even if the input from the user would generate nonsensical output (such as sharing the marginal y axes). I could add a `shareall` keyword to the method that would internally set sharex=True, share all non-diagonal y axes, and warn the user that if they've provided sharex/sharey, those options would be ignored when `shareall=True`... but this does add to the code somewhat, so I can see that not being a desirable option.

It should be noted that under the current PR behaviour, if the user wanted all the joint plots to share a single y axis, they would have to create axes and set up the sharing manually, then pass those axes to arviz. So it's possible but fiddly...

Let me know what your preferred option is and I'll have a go at coding it up.

Reply:
I prefer option 1 which technically would still allow sharex=True or even other stranger configurations provided the user generates the axis grid manually and uses the `ax` argument but makes our code and testing significantly less complex.

Reply:
Right, I've had a go at implementing option 1 - please let me know your thoughts/suggestions/corrections